core/reactor_common.c

```diff
@@ -166,6 +166,88 @@ parse_rti_code_t parse_rti_addr(char* rti_addr);
 void set_federation_id(char* fid);
 #endif
 
+/**
+ * Allocate memory using calloc (so the allocated memory is zeroed out)
+ * and record the allocated memory on the specified self struct so that
+ * it will be freed when calling {@link free_reactor(self_base_t)}.
+ * @param count The number of items of size 'size' to accomodate.
+ * @param size The size of each item.
+ * @param self The self struct on which to record the allocation.
+ */
+void* _lf_allocate(size_t count, size_t size, struct self_base_t *self) {
```

Comment:
I think I don't understand why the third argument is `self_base_t` and not `allocation_record_t`. Would using the latter make this allocation function more generic?